docs/proposals/07-shoot-control-plane-migration.md

```diff
@@ -202,43 +202,99 @@ The ShootState synchronization controller will become part of the gardenlet. It
+*Note:* this resource could aslo be managed by the ETCD Druid and updated depending on the health of the `etcd-backup-restore` sidecar as a more reliable source of when the migration was actually triggered as it fetches this information from a "header" file in the `Shoot`'s backup directory. This would require the `etcd` resource to be deployed as soon as possible in the __Destination Seed__ (before other extension resources) so that the `etcd-backup-restore migrate` operation can be triggered to update the "header" in the `Shoot`'s source backup directory which will cause the `etcd-backup-restore` sidecar in the __Source Seed__ to stop and the `ClusterOwner` resource to be updated before extension controllers in the __Destination Seed__ can start reconciling their resources.
+
+**Alternative: DNS leader election:**
+
+A DNS TXT entry with TTL=60s and value seed='__Source Seed__' is used. The record is created and maintained by the Gardener Controller Manager (by using the DNS Controller Manager and its DNSEntry resource). When a control plane migration is detected, the Gardener Controller Manager changes the value of the DNS Entry to seed='__Destination Seed__' and waits for 2*TTL + 1 = 121 seconds to ensure that the change is propagated to all controllers in the old seed. We rely on the fact that DNS is highly available (100% for AWS Route53) and that control plane components in the __Source Seed__ can see the changes.
+
+Control plane components have to be shut down even when there is no access to the __Source Seed's__ API server. To be able to do that a daemonset is deployed in each Seed cluster. When the daemonset in the __Source Seed__ sees that it is nolonger the leader (by checking the DNS record) and there is no connection to the __Source Seed's__ API server, the daemonset will kill the Shoot's control plane pods by directly talking to the Kubelet's API server. If the __Source Seed's__ API Server comes back up, then gardenlet should take care of scaling down the deployments and statefulsets in the Shoot's control plane. This could be problematic if the gardenlet is in a crashloop backoff or takes too much time to do the scaling.
+
+As an alternative to the daemonset, a sidecar container can be added to each control plane component. The sidecar checks the DNS Entry to see if it is still the leader. If it is not, it shuts down the entire pod. This way we do not run the risk of deploymnets and statefulsets recreating the control plane pods after the seed's apiserver comes back up.
 
-For this approach to work backups must be enabled for the Shoot which is migrated. Additionally, synchronization based on the timestamps in the backup entry depend on the frequency of backups made by the backuprestore container.
+The problems with using DNS as leader election is caching. Additionally, not all DNS servers respect TTL settings.
```

Comment:
Since this is a GEP, albeit in draft state, what's the verdict? Do we have one already?

Reply:
Do you mean whether we chose the DNS or ETCD (object storaged) based leader election? If so, we are going for etcd-based one for now so I can remove the DNS part.